src/GMCPAuthenticator.cpp

```diff
@@ -0,0 +1,110 @@
+/***************************************************************************
+ *   Copyright (C) 2024 by Vadim Peretokin - vperetokin@gmail.com          *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+#include "GMCPAuthenticator.h"
+
+#include "Host.h"
+#include "ctelnet.h"
+#include <QDebug>
+
+GMCPAuthenticator::GMCPAuthenticator(Host* pHost) : mpHost(pHost) {}
+
+void GMCPAuthenticator::saveSupportsSet(const QString& data)
+{
+    auto jsonDoc = QJsonDocument::fromJson(data.toUtf8());
+    auto jsonObj = jsonDoc.object();
+
+    if (jsonObj.contains("type")) {
+        QJsonArray typesArray = jsonObj["type"].toArray();
+        for (const auto& type : typesArray) {
+            mSupportedAuthTypes.append(type.toString());
+        }
+    }
+
+    qDebug() << "Supported auth types:" << mSupportedAuthTypes;
```

Comment:
Might I suggest wrapping these all in:
```suggestion
#if defined(DEBUG_GMCP_AUTHENTICATION)
    qDebug() << "Supported auth types:" << mSupportedAuthTypes;
#endif // DEBUG_GMCP_AUTHENTICATION
```
and then adding the (optional) declarations to the CMake and qmake project files - with a comment like what I started doing with #6285 as I suggested in #6314 ?

Reply:
Um, I meant ALL the `qDebug()` lines - not just ***this one***...!